### ISSUE TITLE
not able to restart with radiation 

### ISSUE THREAD

**Issue by guadabsb15** [BoxLib-Codes/Castro#78]
Not able to restart a run when compiled with radiation. Somehow, it works just  when restarting from the first checkfile (chk00000), but it does not work when restarting from any of the ones created afterwards.

The error is the following:

Regridding radiation object at level 1...
                                         done
341::Assertion `boxes.size() > 0' failed, file "/global/homes/m/mbarrios/BoxLib/Src/C_BaseLib/DistributionMapping.cpp", line 1384 !!!

The Backtrace files obtained with compiling using DEBUG = TRUE, didn't show information about what the problem was

**Comment by zingale** [comment]
okay, I can reproduce this with the 1-d `Rad2TShock` test problem.  Restarting from `chk00000` works fine.  Restarting from a later one (the next output by `inputs.M2.test` is `chk01000`) gives the distribution map error

**Comment by zingale** [comment]
a little more on this: the error occurs in `Radiation::restart`, when reading in the radiation flux register.  This is not an issue for `chk00000` because there is no flux register stored there (as indicated by the lack of the flag `Present` in the `Header` file).

But the latter checkpoint files _do_ have `Present` indicated in the `Header`, indicating we are storing the flux register.

This is where the crash happens

**Comment by zingale** [comment]
Here's the backtrace:

```
=== Please note that the line number reported by addr2line may not be accurate.
    If necessary, one can use 'readelf -wl my_exefile | grep my_line_address'
    to find out the offset for that line.

 0: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4fbbb2]
    BLBackTrace::print_backtrace_info(_IO_FILE*)
    /home/zingale/development/BoxLib//Src/C_BaseLib/BLBackTrace.cpp:86

 1: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4fba9f]
    BLBackTrace::handler(int)
    /home/zingale/development/BoxLib//Src/C_BaseLib/BLBackTrace.cpp:49

 2: ./Castro1d.gnu.DEBUG.MPI.ex() [0x46ba14]
    ParallelDescriptor::Abort(int, bool)
    /home/zingale/development/BoxLib//Src/C_BaseLib/ParallelDescriptor.cpp:207

 3: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4196d2]
    BoxLib::Assert(char const*, char const*, int)
    /home/zingale/development/BoxLib//Src/C_BaseLib/BoxLib.cpp:231

 4: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4440ad]
    DistributionMapping::SFCProcessorMap(BoxArray const&, int)
    /home/zingale/development/BoxLib//Src/C_BaseLib/DistributionMapping.cpp:1386

 5: ./Castro1d.gnu.DEBUG.MPI.ex() [0x440ed4]
    DistributionMapping::define(BoxArray const&, int, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BaseLib/DistributionMapping.cpp:544 (discriminator 4)

 6: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4824b6]
    FabArray<FArrayBox>::defineDoit(BoxArray const&, int, int, FabAlloc, DistributionMapping const*, IntVect const&, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BaseLib/FabArray.H:1711

 7: ./Castro1d.gnu.DEBUG.MPI.ex() [0x47f491]
    FabArray<FArrayBox>::define(BoxArray const&, int, int, FabAlloc, IntVect const&, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BaseLib/FabArray.H:1769

 8: ./Castro1d.gnu.DEBUG.MPI.ex() [0x4c5f60]
    MultiFab::define(BoxArray const&, int, int, FabAlloc, IntVect const&, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BaseLib/MultiFab.cpp:442

 9: ./Castro1d.gnu.DEBUG.MPI.ex() [0x55d7cf]
    FabSet::define(BoxArray const&, int, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BoundaryLib/FabSet.cpp:23

10: ./Castro1d.gnu.DEBUG.MPI.ex() [0x55f56f]
    BndryRegister::define(Orientation, IndexType, int, int, int, int, ParallelDescriptor::Color)
    /home/zingale/development/BoxLib//Src/C_BoundaryLib/BndryRegister.cpp:165

11: ./Castro1d.gnu.DEBUG.MPI.ex() [0x50b4c2]
    FluxRegister::define(BoxArray const&, IntVect const&, int, int)
    /home/zingale/development/BoxLib//Src/C_AmrCoreLib/FluxRegister.cpp:91

12: ./Castro1d.gnu.DEBUG.MPI.ex() [0x50b233]
    FluxRegister::FluxRegister(BoxArray const&, IntVect const&, int, int)
    /home/zingale/development/BoxLib//Src/C_AmrCoreLib/FluxRegister.cpp:25

13: ./Castro1d.gnu.DEBUG.MPI.ex() [0x5d90ab]
    Radiation::restart(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::istream&)
    /home/zingale/development/Castro/Exec/radiation_tests/Rad2Tshock/../../../Source/Radiation/Radiation.cpp:940 (discriminator 3)

14: ./Castro1d.gnu.DEBUG.MPI.ex() [0x59d397]
    Castro::restart(Amr&, std::istream&, bool)
    /home/zingale/development/Castro/Exec/radiation_tests/Rad2Tshock/../../../Source/Castro_io.cpp:404

15: ./Castro1d.gnu.DEBUG.MPI.ex() [0x520830]
    Amr::restart(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    /home/zingale/development/BoxLib//Src/C_AMRLib/Amr.cpp:1473

16: ./Castro1d.gnu.DEBUG.MPI.ex() [0x51ecce]
    Amr::init(double, double)
    /home/zingale/development/BoxLib//Src/C_AMRLib/Amr.cpp:1020

17: ./Castro1d.gnu.DEBUG.MPI.ex() [0x5a1d56]
    main
    /home/zingale/development/Castro/Exec/radiation_tests/Rad2Tshock/../../../Source/main.cpp:123

18: /lib64/libc.so.6(__libc_start_main+0xf1) [0x655f731]
    ??
    ??:0

19: ./Castro1d.gnu.DEBUG.MPI.ex() [0x40afc9]
    _start
    ??:?

```

**Comment by zingale** [comment]
just to confirm, this restart worked with tag 16.10 but fails with tag 16.11

**Comment by zingale** [comment]
more data.  

Code around 2016-10-13 01:00 works
BoxLib: 09fe908919553109b0939a057beecd1f0e803fc9
Castro: 4d2997d14603768b644fe86812e4d7e72ae35f9f

but code around 2016-10-15 12:00 crashes on restart from 1000 with the distribution map error
BoxLib: beb75b32a6c383cb792be0ffbaf84eabf448fefb
Castro: 253a06402c38681d0d5bb5d7f31234647245e4b7

**Comment by zingale** [comment]
doesn't look like there is much in Castro that changed between those, so the culprit is probably something in BoxLib in that small 2-day window.

**Comment by zingale** [comment]
this is fixed with commit 294872cc915ae9718da73bad18013120d72382c9
